### PR TITLE
pkp/pkp-lib#1003 Add custom block manager

### PR DESCRIPTION
**Pull request by mfelczak**
Ported custom block manager from OJS

### CODE REVIEW

File: plugins/generic/customBlockManager/CustomBlockEditForm.inc.php [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+ *
+ * Form for site admins to create and modify sidebar blocks
+ *
+ */
+
+import('lib.pkp.classes.form.Form');
+
+class CustomBlockEditForm extends Form {
+	/** @var $plugin object */
+	var $plugin;
```
Comment (beghelli) [inline plugins/generic/customBlockManager/CustomBlockEditForm.inc.php]: if those members are private, use the underscore $_plugin to name them. I think you have some cases like that below also.

File: plugins/generic/customBlockManager/CustomBlockEditForm.inc.php [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+	}
+
+	/**
+	 * Initialize form data from current group group.
+	 */
+	function initData() {
+		$plugin =& $this->plugin;
+
+		// add the tiny MCE script 
+		$this->addTinyMCE();
```
Comment (beghelli) [inline plugins/generic/customBlockManager/CustomBlockEditForm.inc.php]: we don't have tinyMCE plugin for OHS?

File: plugins/generic/customBlockManager/CustomBlockEditForm.inc.php [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+	function addTinyMCE() {
+		$plugin =& $this->plugin;
+		$templateMgr =& TemplateManager::getManager();
+
+		// Enable TinyMCE with specific params
+		$additionalHeadData = $templateMgr->get_template_vars('additionalHeadData');
+
+		import('classes.file.PublicFileManager');
+		$publicFileManager = new PublicFileManager();
+		$tinyMCE_script = '
```
Comment (beghelli) [inline plugins/generic/customBlockManager/CustomBlockEditForm.inc.php]: I think we can still use CamelCase here...

File: plugins/generic/customBlockManager/CustomBlockManagerPlugin.inc.php [hunk 4]
```diff
@@ -0,0 +1,165 @@
...
+ * @class CustomBlockManagerPlugin
+ *
+ * Plugin to let site admins add and delete sidebar blocks
+ *
+ */
+
+import('lib.pkp.classes.plugins.GenericPlugin');
+
+class CustomBlockManagerPlugin extends GenericPlugin {
+	function getDisplayName() {
```
Comment (beghelli) [inline plugins/generic/customBlockManager/CustomBlockManagerPlugin.inc.php]: missing space line between the class definition and this function

File: plugins/generic/customBlockManager/CustomBlockPlugin.inc.php [hunk 5]
```diff
@@ -0,0 +1,202 @@
...
+ * @class CustomBlockPlugin
+ *
+ * A generic sidebar block that can be customized through the CustomBlockManagerPlugin
+ *
+ */
+
+import('lib.pkp.classes.plugins.BlockPlugin');
+
+class CustomBlockPlugin extends BlockPlugin {
+	var $blockName;
```
Comment (beghelli) [inline plugins/generic/customBlockManager/CustomBlockPlugin.inc.php]: missing doc, extra space line from the class definition

File: plugins/generic/customBlockManager/CustomBlockPlugin.inc.php [hunk 6]
```diff
@@ -0,0 +1,202 @@
...
+	/**
+	 * Determine whether the plugin is enabled. Overrides parent so that
+	 * the plugin will be displayed during install.
+	 */
+	function getEnabled() {
+		if (!Config::getVar('general', 'installed')) return true;
+		return parent::getEnabled();
+	}
+
+	function getManagementVerbs() {
```
Comment (beghelli) [inline plugins/generic/customBlockManager/CustomBlockPlugin.inc.php]: missing doc

File: plugins/generic/customBlockManager/CustomBlockPlugin.inc.php [hunk 7]
```diff
@@ -0,0 +1,202 @@
...
+				__('manager.plugins.enable')
+			);
+		}
+		return $verbs;
+	}
+
+	/**
+	 * Perform management functions
+	 */
+	function manage($verb, $args) {
```
Comment (beghelli) [inline plugins/generic/customBlockManager/CustomBlockPlugin.inc.php]: @copydoc is more appropriate, I think

File: plugins/generic/customBlockManager/CustomBlockPlugin.inc.php [hunk 8]
```diff
@@ -0,0 +1,202 @@
...
+	 */
+	function getBlockContext() {
+		if (!Config::getVar('general', 'installed')) return BLOCK_CONTEXT_RIGHT_SIDEBAR;
+		return parent::getBlockContext();
+	}
+
+	/**
+	 * Determine the plugin sequence. Overrides parent so that
+	 * the plugin will be displayed during install.
+	 */
```
Comment (beghelli) [inline plugins/generic/customBlockManager/CustomBlockPlugin.inc.php]: missing return doc

File: plugins/generic/customBlockManager/CustomBlockPlugin.inc.php [hunk 9]
```diff
@@ -0,0 +1,202 @@
...
+					$form->display();
+				}
+				exit;
+		}
+		return false;
+	}
+
+	/**
+	 * Get the contents of the Block
+	 * @param $templateMgr object
```
Comment (beghelli) [inline plugins/generic/customBlockManager/CustomBlockPlugin.inc.php]: maybe replace object by the template manager class name

File: plugins/generic/customBlockManager/CustomBlockPlugin.inc.php [hunk 10]
```diff
@@ -0,0 +1,202 @@
...
+	 * @return String
+	 */
+	function getDisplayName() {
+		return $this->blockName . ' ' . __('plugins.generic.customBlock.nameSuffix');
+	}
+
+	/**
+	 * Get a description of the plugin.
+	 */
+	function getDescription() {
```
Comment (beghelli) [inline plugins/generic/customBlockManager/CustomBlockPlugin.inc.php]: missing return doc

File: plugins/generic/customBlockManager/SettingsForm.inc.php [hunk 11]
```diff
@@ -0,0 +1,95 @@
+<?php
+
+/**
+ * @file SettingsForm.inc.php
+ *
+ * Copyright (c) 2003-2015 John Willinsky. For full terms see the file docs/COPYING.
+ *
+ * @class SettingsForm
+ *
+ * @brief Form for site admins to add or delete sidebar blocks
```
Comment (beghelli) [inline plugins/generic/customBlockManager/SettingsForm.inc.php]: missing package info, @file should have the complete path, different copyright text

**Comment by beghelli** [comment]
I don't know if you needed a review Michael, but I commented on some problems with code style only.

**Comment by mfelczak** [comment]
Thanks Bruno. This plugin was just copied over from OJS (with warts and all). I'll touch-up some of the formatting and styling.

File: plugins/generic/customBlockManager/CustomBlockEditForm.inc.php (hunk 2 above)
Comment (mfelczak) [inline plugins/generic/customBlockManager/CustomBlockEditForm.inc.php]: We do, but this plugin and the static pages plugin provide their own customized instance.

**Comment by beghelli** [comment]
I imagined that Michael. The code is really old...

**Comment by mfelczak** [comment]
Yep, the OHS core is based off the now quite old ojs-stable-2_3

**Comment by mfelczak** [comment]
Cleaned-up headers, copyright, and function docs

**Pull request merged**